src/main/java/edu/iis/powp/command/gui/ControlsManagerWindow.java

```diff
@@ -118,21 +118,29 @@ public ControlsManagerWindow(Application application) {
     private void moveDownWindow() {
         newPlotter = new MoveDownPlotterDecorator(application.getDriverManager().getCurrentPlotter());
         application.getDriverManager().setCurrentPlotter(newPlotter);
+        DrawerFeature.getDrawerController().clearPanel();
+        commandhistory();
     }
 
     private void moveUpWindow() {
         newPlotter = new MoveUpPlotterDecorator(application.getDriverManager().getCurrentPlotter());
         application.getDriverManager().setCurrentPlotter(newPlotter);
+        DrawerFeature.getDrawerController().clearPanel();
+        commandhistory();
     }
 
     private void moveLeftWindow() {
         newPlotter = new MoveLeftPlotterDecorator(application.getDriverManager().getCurrentPlotter());
```

Comment:
Inconsistent behavior: changing a current driver in application menu deletes current decorators but leaves command history. 
FIX TO MERGE: Attach observer to DriverManager and try to handle it correctly (watch out cyclic notifications!)